src/data.jl

```diff
@@ -255,10 +259,17 @@ function get_target_gate(params::Dict{String, Any}, are_elementary_gates_real::B
     is_target_real = QCO.is_gate_real(params["target_gate"])
 
     if are_elementary_gates_real
-        if !is_target_real
-            Memento.error(_LOGGER, "Infeasible decomposition: all elementary gates have zero imaginary parts")
-        else 
+        if is_target_real
             return real(params["target_gate"]), is_target_real
+        else
+            ref_nonzero_r, ref_nonzero_c = QCO._get_ref_nonzero_index_of_original_target(params["target_gate"])
+            global_phase = angle(params["target_gate"][ref_nonzero_r, ref_nonzero_c])
+            is_target_real_up_to_phase = QCO.is_gate_real(exp(-im*global_phase)*params["target_gate"])
+            if is_target_real_up_to_phase
+                return real(exp(-im*global_phase)*params["target_gate"]), !is_target_real
```

Comment:
Could this also be written as a general evaluation for `global_phase = 0` and then evaluate the global phase of target if `exact_optimal_global_phase` is true? Also, we should add a test to cover this case where target is real in a `global_phase` sense.

Reply:
implemented the global_phase 0, added tests  and including this in the new commit